src/unix/process.c

```diff
@@ -874,14 +872,25 @@ int uv_spawn(uv_loop_t* loop,
   if (err)
     goto error;
 
+#if !defined(__APPLE__)
   uv_signal_start(&loop->child_watcher, uv__chld, SIGCHLD);
+#endif
 
   /* Acquire write lock to prevent opening new fds in worker threads */
   uv_rwlock_wrlock(&loop->cloexec_lock);
 
   /* Spawn the child */
   exec_errorno = uv__spawn_and_init_child(options, stdio_count, pipes, signal_pipe[1], &pid);
 
+#if defined(__APPLE__)
+  if (exec_errorno == 0) {
+    struct kevent event;
+    EV_SET(&event, pid, EVFILT_PROC, EV_ADD | EV_ONESHOT, NOTE_EXIT, 0, 0);
+    if (kevent(loop->backend_fd, &event, 1, NULL, 0, NULL))
+      abort();
+  }
+#endif
```

Comment:
It would probably be better to put this after we release the locks and check the waitpid status for the initial launch (nearer to the bottom of this function)?

Reply:
makes sense. I was thinking that it should be registered as soon as possible after the spawn, because I'm unsure about whether it's possible to miss an event if it happens before you register for it. Is that not the case?